### ISSUE TITLE
nvim-cmp bindings don't work when using Latin layout

### ISSUE THREAD

**Issue by ghost** [Wansmer/langmapper.nvim#8]
[nvim-cmp](https://github.com/hrsh7th/nvim-cmp) bindings like `<C-e>` (abort completion) and `<C-f>` (scroll down through documentation) work perfectly fine in Russian layout but not in English. `<C-*>` bindings set via `nvim_set_keymap()` work as usual.

Tested in https://github.com/Wansmer/nvim-config with only changes being `xkb-switcher` for `get_current_layout_id` and `ru` for `id` of Russian layout.

**Comment by Wansmer** [comment]
In my config (https://github.com/Wansmer/nvim-config) `<C-e>` and `<C-f>` remapped in `lua/mappings.lua`. That's why it no work with `cmp` bindings.

It is a strange, that it is not remapped while `InsertEnter` for `cmp`. I will check it later.

**Comment by Wansmer** [comment]
I've looked at the `nvim-cmp` code, but so far I haven't been able to figure out why this problem occurs.

In addition to the option suggested above, you can disable `hack_keymap` and add `require('langmapper').automapping({ buffer = true })`.

I leave the issue open, because I will still figure it out.

**Comment by edementyev** [comment]
@danilshvalov thanks for the snippet, I did manage to make cmp mappings work modifying it a little bit, here's my version 
```lua
if pcall(require, "langmapper") then
  local keymap = require("cmp.utils.keymap")
  local origin_set_map = keymap.set_map
  local utils = require("langmapper.utils")
  keymap.set_map = function(bufnr, mode, lhs, rhs, opts)
    origin_set_map(bufnr, mode, lhs, rhs, opts)
    origin_set_map(bufnr, mode, utils.translate_keycode(lhs, "default", "ru"), rhs, opts)
  end
end
```

**Comment by danilshvalov** [comment]
As I understand it, the problem is that `nvim-cmp` uses the `vim.api.nvim_buf_get_keymap` function, which returns hacked mappings instead of the necessary ones. So far I have come up with the following hack to solve this problem:

```lua
local utils = require("langmapper.utils")
local original_nvim_buf_get_keymap = vim.api.nvim_buf_get_keymap

vim.api.nvim_buf_get_keymap = function(bufnr, mode)
  local maps = original_nvim_buf_get_keymap(bufnr, mode)
  local filtered = {}
  for _, map in ipairs(maps) do
    if map.lhs == utils.translate_keycode(map.lhs, "default", "ru") then
      table.insert(filtered, map)
    end
  end
  return filtered
end

```

**Comment by danilshvalov** [comment]
`nvim-cmp` has several problems with the `keymap.set_map` substitution solution. So, for example, when pressing `<C-j>` without text, the input can completely freeze. The solution with the substitution of `get_keymap` lacks these disadvantages.

**Comment by Wansmer** [comment]
Fixed with #12, thanks @danilshvalov 

For use with `nvim-cmp`:

```lua
local langmapper = require("langmapper")
langmapper.setup()
langmapper.hack_get_keymap()
```
